airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -135,13 +146,12 @@ def create_nodegroup(
         :rtype: Dict
         """
         eks_client = self.conn
+
         # The below tag is mandatory and must have a value of either 'owned' or 'shared'
         # A value of 'owned' denotes that the subnets are exclusive to the nodegroup.
         # The 'shared' value allows more than one resource to use the subnet.
-        tags = {'kubernetes.io/cluster/' + clusterName: 'owned'}
-        if "tags" in kwargs:
-            tags = {**tags, **kwargs["tags"]}
-            kwargs.pop("tags")
+        tags = kwargs.pop("tags", {})
+        tags[f'kubernetes.io/cluster/{clusterName}'] = 'owned'
```

Comment:
How about changing this function to

```python
def create_nodegroup(
    self,
    clusterName: str,
    nodegroupName: str,
    subnets: List[str],
    nodeRole: str,
    *,
    tags: Optional[dict] = None
    **kwargs,
) -> Dict:
    ...
    if tags is not None:
        resolved_tags = tags
    else:
        resolved_tags = {}
    resolved_tags[f'kubernetes.io/cluster/{clusterName}'] = 'owned'
```

to better leverage Python’s built-in syntax. The `kwargs.pop()` or `if key in kwargs` pattern was a pragmatic workaround on Python 2, but now that keyword-only arguments are available, they make the API both easier to read and write.

Reply:
Thanks, I didn't think of that syntax for the declaration.   If we're going that route, how would you feel about simplifying it to 

```
resolved_tags = tags or {}
resolved_tags[f'kubernetes.io/cluster/{clusterName}'] = 'owned'
```

Reply:
I don’t generally like the `x or default` pattern, but this is OK as well.

Reply:
Resolving this one since we picked up the conversation below in dstandish's thread.